crates/store/re_protos/proto/rerun/v0/catalog.proto

```diff
@@ -0,0 +1,249 @@
+syntax = "proto3";
+
+package rerun.redap.v0;
+
+import "rerun/v0/common.proto";
+
+// The RerunCatalog service is a common API set exposed by:
+// - The Rerun Catalog Server
+// - Any Rerun Storage Node
+//
+// Some Important terminology:
+// - Resource: A resource is any data that can be mapped to a single recording or table in the Rerun data model.
+//   - Examples: A single Rerun recording, an MCAP file, parquet table, etc.
+//   - Every resource across the catalog has a unique identifier.
+// - Collection: A collection contains multiple resources of a similar type.
+//   - Every collection includes a "Property Table" of properties associated with the resources in a collection.
+// - Catalog Entry: An entry within a catalog is a named interface that facilitates access to resources or collections.
+//   - Examples: A collection of recordings, a table of training data, a view created from a query.
+//
+// A Rerun Storage Node generally acts as a single-entry catalog, serving all
+// the recordings stored in the node as a single collection along with a single
+// property-table.  This makes it convenient to directly connect to a Storage Node
+// with the Rerun Viewer in minimal deployments without the need to set up a
+// separate catalog server.
+//
+// This also extends to the Viewer itself, which contains a minimal implementation of a fully
+// in-memory storage node. Each recording sent directly to the viewer from the Rerun SDK will
+// be stored in the viewer's local collection.
+service RerunCatalog {
+    // ---------------- Shared Catalog APIs ------------------
+
+    // List all the Entries in the catalog
+    //
+    // The storage node will only return a single entry for its collection.
+    rpc ListEntries(ListEntriesRequest) returns (ListEntriesResponse) {}
+
+    // ---------------- Catalog-Only Catalog APIs ------------------
+    // These APIs are not implemented in the Storage Node
+
+    // Add a new entry to the catalog
+    rpc AddEntry(AddEntryRequest) returns (AddEntryResponse) {}
+
+    // Remove an entry from the catalog
+    rpc RemoveEntry(RemoveEntryRequest) returns (RemoveEntryResponse) {}
+    // Remove an entry from the catalog
+
+    // ---------------- Shared Collection APIs ------------------
+    // APIs for interacting with catalog entries which are collections
+    //
+    // All of the Request messages include a `catalog_entry` field. This can be omitted
+    // when interacting with a Storage Node.
+
+    // Get the Schema of the property table for a particular catalog entry
+    rpc GetPropertyTableSchema(GetPropertyTableSchemaRequest)
```

Comment:
Nit: Maybe just use `Properties` instead of `PropertyTable`? Or is it important to be explicit about the table aspect here?